RevenueCat/Example/PurchasesListener.cs

```diff
@@ -74,99 +161,404 @@ private void SwitchUser()
+    }
+
+    void LogInAsTest()
+    {
+        LogIn("test");
+    }
+
+    void LogInAsRandomId()
+    {
+        Guid appUserID = Guid.NewGuid();
+        LogIn(appUserID.ToString());
+    }
+
+    void LogIn(string appUserID)
+    {
+        var purchases = GetComponent<Purchases>();
+        purchases.LogIn(appUserID, (customerInfo, created, error) =>
         {
-            Debug.Log("getProducts " + products);
             if (error != null)
             {
                 LogError(error);
             }
+            else
+            {
+                infoLabel.text = $"created: {created}\n customerInfo:\n{customerInfo.ToString()}";
+            }
         });
+    }
 
-        purchases.SyncPurchases();
-        purchases.SetFinishTransactions(false);
-        Debug.Log("user ID " + purchases.GetAppUserId());
-        Debug.Log("user is anonymous " + purchases.IsAnonymous());
+    void LogOut()
+    {
+        var purchases = GetComponent<Purchases>();
+        purchases.LogOut((customerInfo, error) =>
+        {
+            if (error != null)
+            {
+                LogError(error);
+            }
+            else
+            {
+                infoLabel.text = customerInfo.ToString();
+            }
+        });
     }
 
-    private void ButtonClicked(Purchases.Package package)
+    void CheckIntroEligibility()
     {
         var purchases = GetComponent<Purchases>();
-        purchases.PurchasePackage(package, (productIdentifier, customerInfo, userCancelled, error) =>
+        purchases.GetOfferings((offerings, error) =>
         {
-            if (!userCancelled)
+            if (error != null)
             {
-                if (error != null)
+                LogError(error);
+            }
+            else
+            {
+                var productIds = GetPackages(offerings)
+                    .Select(package => package.StoreProduct.Identifier) // map to product ids
+                    .ToArray();
+                purchases.GetProducts(productIds, (storeProducts, innerError) =>
                 {
-                    LogError(error);
+                    if (innerError != null)
+                    {
+                        LogError(innerError);
+                    }
+                    else
+                    {
+                        purchases.CheckTrialOrIntroductoryPriceEligibility(productIds, eligibilitiesByProductId =>
+                        {
+                            var items = eligibilitiesByProductId.Select(kvp =>
+                                string.Format($"{kvp.Key} : Id={kvp.Value.ToString()}"));
+                            infoLabel.text = $"{{ \n {string.Join(Environment.NewLine, items)}\n }} \n ";
+                        });
+                    }
+                });
+            }
+        });
+    }
+
+    void GetPromotionalOffers()
+    {
+        var purchases = GetComponent<Purchases>();
+        purchases.GetOfferings((offerings, error) =>

```

Comment:
decent amount of code duplication on these. I'd love to clean up but as a follow up